File: tests/SingleStmt.pas
namespace Demo;

type
  SingleStmt = public class
  public
    class method TestIf(x: Integer);
    class method TestFor();
  end;

implementation

class method SingleStmt.TestIf(x: Integer);
begin
  if x = 5 then
    System.Console.WriteLine('Hi');
end;

class method SingleStmt.TestFor();
begin
  for i := 1 to 3 do
    System.Console.WriteLine(i);
end;

File: tests/SingleStmt.cs
namespace Demo {
    public partial class SingleStmt {
        public static void TestIf(int x);
        public static void TestFor();
        public static void TestIf(int x) {
            if (x == 5) System.Console.WriteLine('Hi');
        }
        public static void TestFor() {
            for (var i = 1; i <= 3; i++) System.Console.WriteLine(i);
        }
    }
}
